app/script/media/MediaStreamHandler.js

```diff
@@ -316,6 +340,59 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
     return Promise.resolve();
   }
 
+  /**
+   * Check for permission for the requested media type.
+   *
+   * @private
+   * @param {z.media.MediaType} mediaType - Requested media type
+   * @returns {Promise} Resolves when the permissions are not denied
+   */
+  _checkPermissions(mediaType) {
+    if (!z.util.Environment.browser.supports.mediaPermissions) {
+      return Promise.resolve();
+    }
+
+    const permissionTypes = this._getPermissionTypes(mediaType);
+    const shouldCheckPermissions = permissionTypes && permissionTypes.length;
+    return shouldCheckPermissions
+      ? this.permissionRepository.getPermissionStates(permissionTypes).then(permissions => {
```

Comment:
Let's outsource the following bulk into a separate function to make the ternary operator easier to read:

```js
this.permissionRepository.getPermissionStates(permissionTypes).then(permissions => {
          for (const permission of permissions) {
            const {permissionState, permissionType} = permission;
            const isPermissionDenied = permissionState === z.permission.PermissionStatusState.DENIED;
            if (isPermissionDenied) {
              this.logger.warn(`Permission for '${permissionType}' is set to '${permissionState}'`, permissions);
              return Promise.reject(new z.permission.PermissionError(z.permission.PermissionError.TYPE.DENIED));
            }
          }

          return Promise.resolve();
        })
```

Reply:
Good point